src/Core/Abstractions/IStateService.cs

```diff
@@ -13,6 +13,12 @@ namespace Bit.Core.Abstractions
     public interface IStateService
     {
         List<AccountView> AccountViews { get; }
+        Task<UserKey> GetUserKeyAsync(string userId = null);
+        Task SetUserKeyAsync(UserKey value, string userId = null);
+        Task<MasterKey> GetMasterKeyAsync(string userId = null);
+        Task SetMasterKeyAsync(MasterKey value, string userId = null);
+        Task<string> GetUserKeyMasterKeyAsync(string userId = null);
+        Task SetUserKeyMasterKeyAsync(string value, string userId = null);
```

Comment:
I keep going back and forth over how explicit these names need to be. Do you think all the keys should include the word 'Encrypted' then?
- `UserKeyMasterKey` -> `MasterKeyEncryptedUserKey`
- `UserKeyPin` -> `PinKeyEncryptedUserKey`

Reply:
As long as it's consistent across the app, it's fine. However, we need to establish which is encrypting the other, like for a new dev reading `UserKeyMasterKey` for the first time, what does that mean? It's like a combination of both but IMO it doesn't tell the purpose of the relationship between the `UserKey` and the `MasterKey`. In these situations, I like more being explicit; otherwise we need to put somewhere in the docs/comment/summary what's the meaning of that.
@mpbw2 Thoughts?

Reply:
I prefer explicit naming as well citing the same example Fede gave (helpful for new devs, and even experienced devs who haven't thought about the particular code in awhile)

Reply:
Renamed!